### PR TITLE
perf:use sync.RWMutex replace sync.Mutex in counter

### PR DESCRIPTION
**Pull request by liu-song**
**Please provide issue(s) of this PR:
Optimize the use of locks，distinguish between read lock and write lock.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration
- [ ] Docs
- [ ] Installation
- [x] Performance and Scalability
- [ ] Naming
- [ ] HealthCheck
- [ ] Test and Release

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: common/connlimit/listener.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ const (
 type counter struct {
 	size       int32
 	actives    map[string]*Conn // 活跃的连接
-	mu         sync.Mutex
+	mu         sync.RWMutex
```
Comment (chuntaojun) [inline common/connlimit/listener.go:43]: sync.RWMutex => *sync.RWMutex

Comment (liu-song) [reply common/connlimit/listener.go]: Done

**Review by chuntaojun — APPROVED** [review]

**Comment by chuntaojun** [comment]
Add the following issue to associate

**Comment by liu-song** [comment]
https://github.com/polarismesh/polaris/pull/189

**Pull request merged**